### PR TITLE
Auto materialize logic on an external asset graph

### PR DESCRIPTION
**Pull request by johannkm**
Test calling the existing reconciliation sensor logic on an external asset graph

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13362" target="_blank">Graphite</a> 
    * **PR #13380** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13380" target="_blank">Graphite</a> 
      * **PR #13396** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13396" target="_blank">Graphite</a> 
        * **PR #13397** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13397" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13397?utm_source=stack-comment).

File: python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/scenarios.py [hunk 1]
```diff
@@ -309,63 +352,6 @@ def _assets(context):
 # The cases
 ######################################################################
 
-
-class FanInPartitionMapping(PartitionMapping):
```
Comment (johannkm) [inline python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/scenarios.py:313]: these caused some headache- they get ignored when building the ExternalAssetGraph. Replaced with the equivalent StaticPartitionMappings

**Review by sryza — COMMENTED** [review]
Could it make sense to write this in a way that allows us to avoid maintaining duplicate logic between `get_run_requests` and  `reconcile`? Is there a particular reason they need different implementations?

**Comment by johannkm** [comment]
yep, refactored

**Review by sryza — APPROVED** [review]
Nice

**Pull request merged**